### PR TITLE
Build basic Gallery skeleton

### PR DESCRIPTION
**Pull request by rami-a**
This creates 4 new screens that can be built off of to create the new gallery

### CODE REVIEW

File: gallery/lib/demo.dart [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    return Scaffold(
+      appBar: AppBar(
+        title: Text('Demo'),
+      ),
+      body: Center(
+        child: Text('Here is a demo!'),
+      ),
+    );
+  }
+}
```
Comment (johnsonmh) [inline gallery/lib/demo.dart]: Be consistent about newlines at EOF (either always do or always don't)

File: gallery/lib/home.dart [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+      appBar: AppBar(
+        title: Text('Gallery'),
+        leading: IconButton(icon: Icon(Icons.color_lens), onPressed: () {
+          Navigator.of(context).push(
+            MaterialPageRoute(
+              fullscreenDialog: true,
+              builder: (context) => SplashPage(),
+            ),
+          );
+        }),
```
Comment (johnsonmh) [inline gallery/lib/home.dart]: nit: trailing comma and reformat here I think.

File: gallery/lib/home.dart [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        actions: <Widget>[
```
Comment (johnsonmh) [inline gallery/lib/home.dart]: nit: don't need this type.

File: gallery/lib/home.dart [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+          IconButton(icon: Icon(Icons.settings), onPressed: () {
+            Navigator.of(context).push(
+              MaterialPageRoute(
+                fullscreenDialog: true,
+                builder: (context) => SettingsPage(),
+              ),
+            );
+          }),
```
Comment (johnsonmh) [inline gallery/lib/home.dart]: trailing comma here too

File: gallery/lib/home.dart [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+        ],
+      ),
+      body: Padding(
+        padding: const EdgeInsets.all(16.0),
+        child: ListView(
+          children: <Widget>[
```
Comment (johnsonmh) [inline gallery/lib/home.dart]: nit: don't need this type.

File: gallery/lib/home.dart [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+            Text('Gallery', style: Theme.of(context).textTheme.headline,),
+            Container(
+              height: 200,
+              child: ListView(
+                scrollDirection: Axis.horizontal,
+                children: <Widget>[
```
Comment (johnsonmh) [inline gallery/lib/home.dart]: nit: don't need this type.

File: gallery/lib/home.dart [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+                  CarouselCard(title: 'Shrine'),
+                  CarouselCard(title: 'Rally'),
+                  CarouselCard(title: 'Crane'),
+                ],
+              ),
+            ),
+            Text('Categories', style: Theme.of(context).textTheme.headline,),
```
Comment (johnsonmh) [inline gallery/lib/home.dart]: probably don't need the trailing comma, but if you leave it, make sure you run `dartfmt` so this fixes itself.

File: gallery/lib/home.dart [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+
+class CarouselCard extends StatelessWidget {
+  const CarouselCard({Key key, this.title}) : super(key: key);
+
+  final String title;
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsetsDirectional.only(top: 8.0, end: 16.0, bottom: 8.0),
```
Comment (johnsonmh) [inline gallery/lib/home.dart]: You should be consistent about whether or not you include the trailing `.0` on double params like this. My vote is that you should remove it, but either way you should probably be consistent.

File: gallery/lib/demo.dart (hunk 1 above)
Comment (rami-a) [reply gallery/lib/demo.dart]: Done

File: gallery/lib/home.dart (hunk 2 above)
Comment (rami-a) [reply gallery/lib/home.dart]: Done

File: gallery/lib/home.dart (hunk 3 above)
Comment (rami-a) [reply gallery/lib/home.dart]: Done

File: gallery/lib/home.dart (hunk 4 above)
Comment (rami-a) [reply gallery/lib/home.dart]: Done

File: gallery/lib/home.dart (hunk 5 above)
Comment (rami-a) [reply gallery/lib/home.dart]: Done

File: gallery/lib/home.dart (hunk 6 above)
Comment (rami-a) [reply gallery/lib/home.dart]: Done

File: gallery/lib/home.dart (hunk 7 above)
Comment (rami-a) [reply gallery/lib/home.dart]: Done

File: gallery/lib/home.dart (hunk 8 above)
Comment (rami-a) [reply gallery/lib/home.dart]: Keeping the `.0` on all, I think that will be the style guide for this project.

File: gallery/lib/home.dart [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+class CarouselCard extends StatelessWidget {
+  const CarouselCard({Key key, this.title}) : super(key: key);
+
+  final String title;
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsetsDirectional.only(
+        top: 8.0,
```
Comment (clocksmith) [inline gallery/lib/home.dart]: drop decimal?

File: gallery/lib/home.dart [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text('Gallery'),
+        leading: IconButton(
+          icon: Icon(Icons.color_lens),
+          onPressed: () {
+            Navigator.of(context).push(
+              MaterialPageRoute(
+                fullscreenDialog: true,
```
Comment (clocksmith) [inline gallery/lib/home.dart]: why are these dialogs?

File: gallery/lib/home.dart (hunk 9 above)
Comment (rami-a) [reply gallery/lib/home.dart]: Choosing to keep it and make that the style for this project

File: gallery/lib/home.dart (hunk 10 above)
Comment (rami-a) [reply gallery/lib/home.dart]: This flag actually makes the route slide up from the bottom and have an X in the app bar, not really a dialog in the traditional sense

**Pull request merged**